Add readme (#26)
patrickelectric/cpy-rs: README.md

This project allows you to create a multibinding export core, making easier for developers to create and distribute libraries that are already made in Rust, but now making them accessible to a broader range of users.

This README provides essential information on how to get started with cpy-rs, its key features, installation guidelines, and usage examples.

## Features

- Easy deployment of libraries to Python and C++ environments.
- Simplified multibinding export core for seamless integration.
- Empowers developers to distribute Python and C++ libraries with minimal effort.
- Supports a wide range of use cases, making it versatile for different projects.

## Getting Started

### Prerequisites

Before using cpy-rs, ensure you have the following prerequisites installed on your system:

- Rust 
- Python
- C++ compiler

### Installation

To use cpy-rs in your project, run `cargo add cpy-binder`, or include it as a dependency in your Rust project's `Cargo.toml` file:

```toml
[dependencies]
cpy-binder = "0.3.0"

### Usage

1. Import the `cpy-binder` macro into your Rust project:

```rust
#[cfg(feature = "python")]

> use pyo3::{prelude::*, wrap_pyfunction};

Reviewer: This include is not necessary.

use cpy_binder::export_cpy;
```

2. Initiate your Rust module collections like following.

```rust
// Rust function using cpy macro
export_cpy!(
    mod my_module {
        fn return_true() -> bool {

>             let x = true

Reviewer: This code is wrong, is not returning at all

        }

>     }

Reviewer: This code is not closing the scope of export_cpy

```

3. Build your Rust project with the `cpy` feature enabled:

```bash

> cargo build

Reviewer: It's missing the build script for binding generation for C/C++

> cargo build

Reviewer: It's missing toolchain configuration for nightly build, project do not build with current stable.

```

4. Your Rust functions are now accessible from Python and C++!

For detailed examples and more advanced usage, please refer to the [Examples](https://github.com/patrickelectric/cpy-rs/tree/main/example) section.

## Examples

Here's a quick sneak peek at a Python and C++ code using a Rust function exported with `cpy-rs`:

```python
# Python code
import my_module

result = my_module.return_true()
print(result)
```

```cpp
// C++ code
#include <iostream>

> #include "my_module.cpp"

Reviewer: This include is totally wrong. And as said, it's missing binding generation for the header.

int main() {

>     bool result = my_rust_module::return_true();

Reviewer: This is wrong, the cbindgen does not generate the function as namespace.

# cpy-rs: Python and C++ Multibinding Export Macro

> ![cpy-rs Logo](https://github.com/patrickelectric/cpy-rs/blob/main/cpy-rs-logo.png)

Reviewer: You forgot to add the image.

- [Features](#features)
- [Getting Started](#getting-started)
  - [Prerequisites](#prerequisites)
  - [Installation](#installation)
  - [Usage](#usage)
- [Examples](#examples)
- [Contributing](#contributing)
- [License](#license)

## Introduction

cpy-rs is a powerful macro that simplifies the deployment of libraries to both Python and C++. 
This project allows you to create a multibinding export core, making easier for developers to create and distribute libraries that are already made in Rust, but now making them accessible to a broader range of users.

This README provides essential information on how to get started with cpy-rs, its key features, installation guidelines, and usage examples.

## Features

- Easy deployment of libraries to Python and C++ environments.
- Simplified multibinding export core for seamless integration.
- Empowers developers to distribute Python and C++ libraries with minimal effort.
- Supports a wide range of use cases, making it versatile for different projects.

## Getting Started

### Prerequisites

Before using cpy-rs, ensure you have the following prerequisites installed on your system:

- Rust 
- Python
- C++ compiler

### Installation

> To use cpy-rs in your project, you can include it as a dependency in your Rust project's `Cargo.toml` file:
> 
> ```toml
> [dependencies]
> cpy-binder = "0.3.0"
> ```

Reviewer 2: ```suggestion
To use cpy-rs in your project, run `cargo add cpy-binder`, or include it as a dependency in your Rust project's `Cargo.toml` file:

```toml
[dependencies]
cpy-binder = "0.3.0"
```
```

### Usage

1. Import the `cpy-binder` macro into your Rust project:

```rust
#[cfg(feature = "python")]
use pyo3::{prelude::*, wrap_pyfunction};

use cpy_binder::export_cpy;
```

2. Initiate your Rust module collections like following.

```rust
// Rust function using cpy macro
export_cpy!(
    mod my_module {
        fn return_true() -> bool {
            let x = true
        }
    }
```

3. Build your Rust project with the `cpy` feature enabled:

```bash
cargo build
```

4. Your Rust functions are now accessible from Python and C++!

For detailed examples and more advanced usage, please refer to the [Examples](https://github.com/patrickelectric/cpy-rs/tree/main/example) section.

## Examples

> Here's a quick sneak peek at a Python and C++ code using a Rust function exported with `cpy-rs`:

Reviewer 2: ba dum tss
